### ISSUE TITLE
Unable to change static IP

### ISSUE THREAD

**Issue by deanfourie1** [esphome/issues#4815]
### The problem

I have currently assigned a static IP to all my devices.

When I try to change the static IP on a device, and do a upload, it immediately tries to connect to the new static IP to do the upload, instead of the old IP, resulting in a failed upload.

### Which version of ESPHome has the issue?

2023.8.1

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

2022.8.3

### What platform are you using?

ESP32

### Board

esp32dev

### Component causing the issue

wifi/networking

### Example YAML snippet

```yaml
manual_ip:
    static_ip: 172.16.102.13
    gateway: 172.16.102.1
    subnet: 255.255.255.224
```

### Additional information

Basically, I cant change the static IP as it just tried to connect to the device on the NEW ip when doing the upload rather then the OLD ip.

Is there a way to change the static IP via OTA?

**Comment by randybb** [comment]
Just change that address in your config and temporary set the old address via `use_address:`. After OTA remove that `use_address:` from your config.
https://esphome.io/components/esphome.html?highlight=use_address#changing-esphome-node-name
